app/controllers/api/messaging.php

```diff
@@ -0,0 +1,2138 @@
+    ->param('name', '', new Text(128), 'Provider name.')
+    ->param('default', false, new Boolean(), 'Set as default provider.', true)
+    ->param('enabled', true, new Boolean(), 'Set as enabled.', true)
+    ->param('apiKey', '', new Text(0), 'Sendgrid API key.')
+    ->inject('dbForProject')
+    ->inject('response')
+    ->action(function (string $providerId, string $name, bool $default, bool $enabled, string $apiKey, Database $dbForProject, Response $response) {
+        $providerId = $providerId == 'unique()' ? ID::unique() : $providerId;
+        $provider = new Document([
+            '$id' => $providerId,
+            'name' => $name,
+            'provider' => 'sendgrid',
+            'type' => 'email',
+            'default' => $default,
+            'enabled' => $enabled,
+            'options' => [],
+            'search' => $providerId . ' ' . $name . ' ' . 'sendgrid' . ' ' . 'email',
+            'credentials' => [
+                'apiKey' => $apiKey,
+            ],
+        ]);
+
+        // Check if a default provider exists, if not, set this one as default
+        if (
+            empty($dbForProject->findOne('providers', [
+                Query::equal('default', [true]),
+                Query::equal('type', ['sms'])
+            ]))
+        ) {
+            $provider->setAttribute('default', true);
+        }
+
+        try {
+            $provider = $dbForProject->createDocument('providers', $provider);
+        } catch (DuplicateException) {
+            throw new Exception(Exception::PROVIDER_ALREADY_EXISTS);
+        }
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_CREATED)
+            ->dynamic($provider, Response::MODEL_PROVIDER);
+    });
+
+App::post('/v1/messaging/providers/msg91')
+    ->desc('Create Msg91 Provider')
+    ->groups(['api', 'messaging'])
+    ->label('audits.event', 'providers.create')
+    ->label('audits.resource', 'providers/{response.$id}')
+    ->label('scope', 'providers.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN, APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'messaging')
+    ->label('sdk.method', 'createMsg91Provider')
+    ->label('sdk.description', '/docs/references/messaging/create-msg91-provider.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_PROVIDER)
+    ->param('providerId', '', new CustomId(), 'Provider ID. Choose a custom ID or generate a random ID with `ID.unique()`. Valid chars are a-z, A-Z, 0-9, period, hyphen, and underscore. Can\'t start with a special char. Max length is 36 chars.')
+    ->param('name', '', new Text(128), 'Provider name.')
+    ->param('default', false, new Boolean(), 'Set as default provider.', true)
```

Comment:
what if i set multiple providers to be default? 👀

Reply:
@fanatic75 I don't think we actually want this parameter in the API 🤔 just in the DB

Reply:
How would one update the default provider then? 🧐